### PR TITLE
Designed diary card

### LINKED ISSUE #15: Diary entry
Variants:
- Expanded
- Collapsed
- New

### LINKED ISSUE #19: Diary
Contains components:
- Diary entry
- Navbar

### PR DESCRIPTION
**Pull request by jonbjarki**
Finished designing diary card

### CODE REVIEW

**Review by boathungry — APPROVED** [review]
👯‍♂️

File: src/app/[locale]/diary/page.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { cookies } from 'next/headers';
+import DiarySidebar from '@/components/diary/DiarySidebar';
+import DiaryList from '@/components/diary/DiaryList';
+
+export default async function Diary() {
+  // Require User to be logged in
+  const supabase = createServerComponentClient({ cookies });
+  const {
+    data: { user },
+  } = await supabase.auth.getUser();
```
Comment (PatrekurTh) [inline src/app/[locale]/diary/page.tsx:13]: Faster to use getSession() for this particular use case.

File: src/app/[locale]/diary/page.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  if (!user) return redirect('/login');
+
+  // Fetches diary entries from database
+  const { data, error } = await supabase
+    .from('diary')
+    .select('*,emotion (name, intensity),symptoms (name)')
+    .eq('user_id', user?.id);
```
Comment (PatrekurTh) [inline src/app/[locale]/diary/page.tsx]: Can be enforced with RLS and skip the equality check

File: src/components/article/ArticleSidebar.tsx [hunk 3]
```diff
@@ -21,7 +22,14 @@ export default async function ArticleSidebar() {
       </div>
       <nav className="flex flex-col">
         {data.map((article) => (
-            <Link key={article.id} className='w-full h-full min-h-16 pl-4 text-lg flex items-center  hover:bg-lm-light' prefetch href={`/reading/${article.id}`}>{article.title}</Link>
+          <Link
+            key={article.id}
+            className="w-full h-full min-h-16 pl-4 text-lg flex items-center  hover:bg-lm-light"
+            prefetch
```
Comment (PatrekurTh) [inline src/components/article/ArticleSidebar.tsx]: prefetch became opt-out in version 9 (deprecated)

File: src/components/diary/DiaryCard.tsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  coping_strategies,
+  behaviour,
+}: DiaryCardProps) {
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  const toggleExpand = () => {
+    setIsExpanded(!isExpanded);
+  };
+
+  const options = isExpanded ? 'flex-col' : 'flex-row';
```
Comment (PatrekurTh) [inline src/components/diary/DiaryCard.tsx:41]: You can use clsx here

File: src/components/diary/DiaryList.tsx [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+export default async function DiaryList({ entries }) {
+  const [diaryEntries, setDiaryEntries] = useState(entries);
+  const [addingEntry, setAddingEntry] = useState(false);
+
+  const supabase = createClientComponentClient<Database>();
+
+  const { data, error } = await supabase.storage.from('images').list('mood', {
+    limit: 100,
+  });
+  console.log(data);
```
Comment (PatrekurTh) [inline src/components/diary/DiaryList.tsx:17]:
```suggestion
```

**Pull request merged**